source/extensions/filters/http/on_demand/on_demand_update.cc

```diff
@@ -0,0 +1,62 @@
+#include "extensions/filters/http/on_demand/on_demand_update.h"
+
+#include "common/common/assert.h"
+#include "common/common/enum_to_int.h"
+#include "common/http/codes.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace OnDemand {
+
+Http::FilterHeadersStatus OnDemandRouteUpdate::requestRouteConfigUpdate() {
+  if (callbacks_->route() != nullptr || !callbacks_->canRequestRouteConfigUpdate()) {
+    return Http::FilterHeadersStatus::Continue;
+  } else {
+    callbacks_->requestRouteConfigUpdate([this]() -> void { onRouteConfigUpdateCompletion(); });
```

Comment:
this could possibly be applied to other onDemandXDS as well: once this requestRouteConfigUpdate is fired, this httpRequest is hung. if there is a problem w/ the RDS subscription, say it's teared down or management server is down, these httpRequests are essentially leaked. 
We should consider either adding some time-out mechanism here, or creating some stats on which we could build alerts. WDYT?

Reply:
Hmm. Good question, I need to think about this.

The latest changes got rid of lambda callbacks, and rely on an interface (which is implemented by ActiveStream). I need to verify this, but I suspect ActiveStream timeout will take care of hung on-demand requests.

Definitely a good idea to collect stats on this.

Reply:
I think we are re: cleanup on timeouts. ActiveStream has a few timeout timers (request timeout handling is set up here: https://github.com/envoyproxy/envoy/blob/master/source/common/http/conn_manager_impl.cc#L531). Stats are being collected for timeouts too, but perhaps it would be useful to have OnDemand filter-specific stats (timeouts would be hard, but I can keep counts of sent requests/received updates), WDYT?